install/universal_pkg/setup.py

```diff
@@ -0,0 +1,235 @@
+import ctypes
+import pkg_resources
+import os
+import sys
+from typing import List, Optional
+
+from setuptools import setup
+
+
+VERSION = '10.4.0'
+
+# List of packages supported by this version of CuPy.
+PACKAGES = [
+    'cupy-cuda102',
+    'cupy-cuda110',
+    'cupy-cuda111',
+    'cupy-cuda112',
+    'cupy-cuda113',
+    'cupy-cuda114',
+    'cupy-cuda115',
+    'cupy-cuda116',
+    'cupy-rocm-4-0',
+    'cupy-rocm-4-2',
+    'cupy-rocm-4-3',
+    'cupy-rocm-5-0',
+]
+
+# List of packages NOT supported by this version of CuPy.
+PACKAGES_OUTDATED = [
+    'cupy-cuda80',
+    'cupy-cuda90',
+    'cupy-cuda91',
+    'cupy-cuda92',
+    'cupy-cuda100',
+    'cupy-cuda101',
+]
+
+# List of sdist packages.
+PACKAGES_SDIST = [
+    'cupy',
+]
+
+
+class AutoDetectionFailed(Exception):
+    pass
+
+
+def _log(msg: str) -> None:
+    sys.stdout.write(f'[cupy-wheel] {msg}\n')
+    sys.stdout.flush()
+
+
+def _get_version_from_library(
+        libnames: List[str], funcname: str) -> Optional[int]:
+    """Returns the library version from list of candidate libraries."""
+
+    for libname in libnames:
+        try:
+            _log(f'Looking for library: {libname}')
+            runtime_so = ctypes.CDLL(libname)
+            break
+        except Exception as e:
+            _log(f'Failed to open {libname}: {e}')
+            return None
```

Comment:
```suggestion
            return None
```

This will cause the first one failing to return an error.
If we pass multiple I think we don't want this.

Reply:
Thanks!